### ISSUE TITLE
copyKernels should be enabled with ZFS or else grub can break

### ISSUE THREAD

**Issue by jb55** [NixOS/nixpkgs#60902]
I had an "oh shi" moment recently where every entry in my grub boot member failed to load and I couldn't boot into nixos, only getting this message:

```
Error: external pointer tables not supported
Error: you need to load the kernel first.
```

after perusing the old nix-dev mailing list I ran into a message from 2017 by @spinus where he ran into the same issue with ZFS.

 I ended up fixing it by booting with a usb-stick and re-installing with `boot.loader.grub.copyKernels = true` which fixed it.

A response by @CMCDragonkai explains:

```
Hey glad you solvee it. So the problem was that you were using Grub to
directly access ZFS root drive to get the kernel, but Nixos uses lots
of hardlinks and Grub has a problem reading that. So the kernel is now
copied to /boot. That's actually how it's done in gummiboot and
probably other bootloaders.
```

Not sure how common this edgecase is but it's terrible when you hit it. Perhaps this option can be turned on when using ZFS?

**Comment by alyssais** [comment]
I agree this should be on by default. You’re not the first person I’ve seen bitten by this. cc @spacekookie

**Comment by Atemu** [comment]
Is this still an issue? When I still used ZFS, I never had to set the option.

**Comment by jb55** [comment]
It's the only time my nixos install has ever been bricked so I keep this option on.

**Comment by Atemu** [comment]
@jb55 could you `nixos-rebuild build` your config and see whether it changes with the config set and not?

**Comment by jb55** [comment]
no I do not want to break my machine

**Comment by jb55** [comment]
oh just a build, let me check

**Comment by jb55** [comment]
yes it does change when I switch it to false
